### ISSUE TITLE
[Bug Report][3.6.8] (VTreeview) treeview treating empty children array differently to vuetify ^2.5.8 

### ISSUE THREAD

**Issue by mcc0rmack** [vuetifyjs/vuetify#19983]
### Environment
**Vuetify Version:** 3.6.8
**Last working version:** ^2.5.8
**Vue Version:** 3.4.27
**Browsers:** Edge 126.0.0.0
**OS:** Windows 10

### Steps to reproduce
As per link, previously empty children array was ignored.

Now it prevents slots from firing, so icons etc no longer work

### Expected Behavior
empty children array is ignored and treated as having no children.

### Actual Behavior
thinks there are children, shows tree arrow and cannot fire slots

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNqtlcFuozAQhl/Fy4VWKuS0F5RmVWl7XK201yRaOfZA3RgbgUkTRbz7jk0gdko2rVQOgP+Z+WbssWF5jJqazZ6qKt21EGXR3EBZSWpgsVKEzHeJqQF2At7skJBdUmoO8nEVWX0V9WomMKpB0T1HVVeggFtZCSOolIffqAxm65ts4YB2RcuRRZkRO2roRkIvNCCBeWNLTbRKmBRsayVXKdY6VI5FNlKbrKoBXW3+o0v24EJJt4pOEW5+gmmFESJHv2/WLc0Fpjr7EHI89pE/SFxykeRacqgTK8Uk87WYdN2InvXsiVwgG1hYqs3ULMekawy/jJvPwoagdegICoEVhw2rRWWcJ+wrXRvCIaetNOTY4ziubEbu7snjgtydNGyF35+MLOOq3eDqxuuHwcNVmg0Ue72YUp4mL6kqWlpAYrXv8RhEyCvGOBeFu+a1CU0aUzkjQ2Nix7695Jf0ktZbrt8Cr4rnQwewwgSHyUbvAw9V+B6iRJRvN3vj27lmbQnKJLo1UqjAdS+H6ThX2DOQZ3s3vtkG4SqeV88dC1S8LXV+RX9hJAbEaSGMn48Q9iIkr11PzjQ/1RWSXe6/eFJb7NrXEE874hrLly8IPqUx1FxS/kuapPlEqQudYo/fM4dta4uftvuT7i9/SaY8rk4spzt7aFO8vc90vYoP84XisE/t+bqKnzKG/E813G5GUShdB0dgzIbHxtdv4jZ0AzLFJcpFkYZfgpEZyre3JGVbPMxIC78cHi803CT+eX76+es5LfkkLpRvwvB/+rXzPdBapVKz7ecaMnS9u3cv+IOaz8b/RNQ9qFZKd1v/Ay3vK8E=)

**Comment by robertzak** [comment]
can this be tagged with `C:VTreeview`? I was going to put in this ticket as well and almost missed someone else had because it wasn't showing up when I filtered to look at the active issues for VTreeview.

Also I second that this be fixed. It looks like it used to be a problem in vuetify 2.x as well and was fixed for this issue: https://github.com/vuetifyjs/vuetify/issues/5357

**Comment by KaelWD** [comment]
Duplicate of #18172

**Comment by robertzak** [comment]
So is the expectation that even if we don't provide a loadChildren function an empty array will still be rendered as though children are present in the tree? Is the fix for this to make sure any children are `undefined` rather than an empty array? I've been waiting on this (and several other fixes) to the tree so we can finish our upgrade to vue3

**Comment by KaelWD** [comment]
Yes. You can also pass a custom itemChildren function if it isn't possible to change the items structure:

```
:item-children="item => item.children.length ? item.children : null"
```

**Comment by robertzak** [comment]
OK. Thank you. I'll give the function a try before trying to rewrite the tree data.
